docs/data/joy/components/select/SelectUsage.js

```diff
@@ -11,11 +11,20 @@ export default function SelectUsage() {
       componentName="Select"
       data={[
         {
+          formLabel: 'Select variant',
           propName: 'variant',
           knob: 'radio',
           defaultValue: 'outlined',
           options: ['plain', 'outlined', 'soft', 'solid'],
         },
+        {
```

Comment:
It would be great if You could add the option to select options color as well. Right now it's always defaults to neutral.

Reply:
Added! https://deploy-preview-36843--material-ui.netlify.app/joy-ui/react-select/

Reply:
Thanks for that! 
I have one more thing for you to consider. Right now if I want to options follow the same variant/color as select and listbox, you'll need to set them explicitly on every option:
![image](https://github.com/mui/material-ui/assets/97035907/e7e99eef-f84d-4ec9-9b0d-813f4815bb42)

What if options could inherit variant/color from the listbox if not specified explicitly? I would use react context to pass thier values from listbox to options. 
If Option component is not used inside Select component their value can fallback to default neutral values.

Reply:
Though it'd be good to have a demo for `<Option>` colors, I truly feel that this might not the best way to showcase it. We ended up disrupting the component demo and lacking of contrast when used by different colors and variants on select and the list—something that's feasible but not recommended without tweaks to the colors on the theme. That said, I truly don't like the idea of a demo that shows something that we don't recommend.

<img width="837" alt="Screenshot 2023-07-20 at 08 47 43" src="https://github.com/mui/material-ui/assets/37222944/c1f3b7c3-c531-442b-b449-3c7a8ff6302f">

Reply:
> What if options could inherit variant/color from the listbox if not specified explicitly?

IMHO this is a good idea about this component behavior, the list should inherit the `Select` style by default, and only override when, as you've said, specified explicitly on the `<Option>`. This would also help with the first demo because we could only show the `select variant` and `select color` options—and then coming up with a separate demo just to show how to custom the `<Option>`